pystan/misc.py

```diff
@@ -462,7 +462,8 @@ def _config_argss(chains, iter, warmup, thin,
         inits_specified = True
     if not inits_specified and isinstance(init, Callable):
         ## test if function takes argument named "chain_id"
-        if "chain_id" in inspect.getargspec(init).args:
+        getfullargspec = inspect.getfullargspec if hasattr(inspect, "getfullargspec") else inspect.getargspec
```

Comment:
Why is this being changed? Is this a compatibility thing?

Reply:
Yes, python 3 vs python 2. Assuming we start to move pystan3 soon, then this wont break stuff in future.

Reply:
Please correct me if I'm wrong. It seems like this has nothing to do with updating pystan to Stan 2.22.1.

I think it would be better if these sorts of things were raised for discussion as issues or distinct pull requests.

Reply:
True, let's discuss this in another PR /issue.